### PR TITLE
UF-6255: Add shedlock functionallity

### PR DESCRIPTION
**Pull request by josc04**
- Added database functionallity
- Added flyway
- Added shedlock functionallity
- Minor code refactoring and formatting

### CODE REVIEW

File: src/test/java/se/sundsvall/citizenchanges/util/NumberFormatterTest.java [hunk 1]
```diff
@@ -5,16 +5,16 @@
...
 
-	private Pattern pattern;
+	private static Pattern PATTERN;
 
-	@BeforeEach
-	void setup() {
-		pattern = Pattern.compile("^\\+[1-9]\\d{3,14}$");
+	@BeforeAll
+	static void setup() {
+		PATTERN = Pattern.compile("^\\+[1-9]\\d{3,14}$");
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/citizenchanges/util/NumberFormatterTest.java:17]: Regexp could be a constant with a name that descriibes the purpose.
I.g. `private static final String REG_EXP_VALID_MOBILE_NUMBER = "^\\+[1-9]\\d{3,14}$";`

**Comment by josc04** [comment]
Acted on PR comment and adjusted constant name

**Review by foobar-1337 — DISMISSED** [review]

File: src/test/java/se/sundsvall/citizenchanges/ShedlockAnnotationsTest.java
Comment (mha109) [inline src/test/java/se/sundsvall/citizenchanges/ShedlockAnnotationsTest.java:1]: What on earth is this monstrosity? Are you trying to impress someone with the unnecessary use of var everywhere? It's like you're on a mission to make your code as unreadable as possible.

Let's talk about this verifyMandatorySchedlockAnnotations method. It's a maze of streams, maps, and flatMaps. Do you enjoy making simple things complicated? This code screams "I know functional programming buzzwords" without actually making the code better.

The hasEnableSchedulerLock method is equally confusing. You're doing this convoluted dance with streams and lambdas just to check if there's an annotation present. It's like you're trying to win a code golf competition.

And what's the deal with getMethodsAnnotatedWith? It's like you've reinvented the wheel for scanning methods with a specific annotation. There are libraries and frameworks that can do this more efficiently. Did you just want to show off your knowledge of reflection?

And please, spare me the unnecessary formatting strings in your error messages. It doesn't make them more readable; it just adds clutter.

In conclusion, simplicity and clarity should be your goals. Stop trying to showcase your functional programming and reflection skills. Write code that others can actually understand without pulling their hair out.

Comment (mha109) [inline src/test/java/se/sundsvall/citizenchanges/ShedlockAnnotationsTest.java:1]: What on earth is this monstrosity? Are you trying to impress someone with the unnecessary use of var everywhere? It's like you're on a mission to make your code as unreadable as possible.

And what's with these overly verbose method names? Do you think adding the word "verify" in every method name makes your code more robust? It just adds noise.

Let's talk about this verifyMandatorySchedlockAnnotations method. It's a maze of streams, maps, and flatMaps. Do you enjoy making simple things complicated? This code screams "I know functional programming buzzwords" without actually making the code better.

The hasEnableSchedulerLock method is equally confusing. You're doing this convoluted dance with streams and lambdas just to check if there's an annotation present. It's like you're trying to win a code golf competition.

And what's the deal with getMethodsAnnotatedWith? It's like you've reinvented the wheel for scanning methods with a specific annotation. There are libraries and frameworks that can do this more efficiently. Did you just want to show off your knowledge of reflection?

And please, spare me the unnecessary formatting strings in your error messages. It doesn't make them more readable; it just adds clutter.

File: src/main/java/se/sundsvall/citizenchanges/configuration/ShedLockConfiguration.java
Comment (mha109) [inline src/main/java/se/sundsvall/citizenchanges/configuration/ShedLockConfiguration.java:1]: What's going on with this ShedLockConfiguration? It's like you're trying to obfuscate a simple configuration. Why are you using @Configuration for such a trivial setup? Are you trying to make a big deal out of a lock provider bean?

And the lockProvider method – why are you passing a DataSource as a parameter? Why not just inject it directly using @Autowired like a normal person? It's not like you have a choice of different data sources here.

The usage of JdbcTemplateLockProvider is fine, but why are you building its configuration like it's some kind of complex rocket science experiment? You're not launching a satellite; you're configuring a lock provider. Keep it simple.

And the use of usingDbTime() – seriously? You're using the database time for a lock provider configuration? Is this a lock provider or a time machine? Keep it practical and straightforward.

In conclusion, there's no need to overcomplicate a simple bean configuration. Make it clear, concise, and avoid unnecessary complexity. Remember, simplicity is the ultimate sophistication.

**Review by mha109 — APPROVED** [review]
